File: src/main/java/io/tatum/model/response/ledger/Customer.java
package io.tatum.model.response.ledger;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import io.tatum.model.request.Country;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * The type Customer.
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@EqualsAndHashCode
@JsonIgnoreProperties(ignoreUnknown = true)
public class Customer {

    private String id;

    private String externalId;

    private Country customerCountry;

    private Fiat accountingCurrency;

    private Country providerCountry;

    private Boolean active;

    private Boolean enabled;
}

File: src/main/kotlin/model/response/ledger/Customer.kt
package model.response.ledger

import model.request.Country

class Customer(
    val id: String,
    val externalId: String,
    val customerCountry: Country? = null,
    val accountingCurrency: Fiat? = null,
    val providerCountry: Country?,
    val active: Boolean,
    val enabled: Boolean,
)
